packages/rainbowkit/src/components/RainbowKitProvider/AuthenticationContext.tsx

```diff
@@ -77,6 +82,24 @@ export function RainbowKitAuthenticationProvider<Message = unknown>({
     }
   }, [status, adapter, isDisconnected]);
 
+  const handleChangedAccount = ({ account }: ConnectorData) => {
+    // Only if account is changed then signOut
+    if (account) adapter.signOut();
+  };
+
+  // biome-ignore lint/nursery/useExhaustiveDependencies: <explanation>
+  useEffect(() => {
```

Comment:
Would add more brevity to comments here to describe exactly why listeners in `useEffect` are more reliable than `onConnect`

Reply:
Yup adding it 🙏